presto-jdbc/src/main/java/com/facebook/presto/jdbc/AbstractConnectionProperty.java

```diff
@@ -0,0 +1,166 @@
+        this.isRequired = requireNonNull(isRequired, "isRequired is null");
+        this.isAllowed = requireNonNull(isAllowed, "isAllowed is null");
+        this.converter = requireNonNull(converter, "converter is null");
+    }
+
+    protected AbstractConnectionProperty(
+            String key,
+            Predicate<Properties> required,
+            Predicate<Properties> allowed,
+            Converter<T> converter)
+    {
+        this(key, Optional.empty(), required, allowed, converter);
+    }
+
+    @Override
+    public String getKey()
+    {
+        return key;
+    }
+
+    @Override
+    public Optional<String> getDefault()
+    {
+        return defaultValue;
+    }
+
+    @Override
+    public DriverPropertyInfo getDriverPropertyInfo(Properties mergedProperties)
+    {
+        String currentValue = mergedProperties.getProperty(key);
+        DriverPropertyInfo result = new DriverPropertyInfo(key, currentValue);
+        result.required = isRequired.test(mergedProperties);
+        return result;
+    }
+
+    @Override
+    public boolean isRequired(Properties properties)
+    {
+        return isRequired.test(properties);
+    }
+
+    @Override
+    public boolean isAllowed(Properties properties)
+    {
+        return !properties.containsKey(key) || isAllowed.test(properties);
+    }
+
+    @Override
+    public Optional<T> getValue(Properties properties)
+            throws SQLException
+    {
+        String value = properties.getProperty(key);
+        if (value == null) {
+            if (isRequired(properties)) {
+                throw new SQLException(format("Connection property '%s' is required", key));
+            }
+            return Optional.empty();
+        }
+        if (value.isEmpty()) {
+            throw new SQLException(format("Connection property '%s' value is empty", key));
+        }
+
+        try {
+            return Optional.of(converter.convert(value));
+        }
+        catch (RuntimeException e) {
+            throw new SQLException(format("Connection property '%s' value is invalid: %s", key, value), e);
+        }
+    }
+
+    @Override
+    public void validate(Properties properties)
+            throws SQLException
+    {
+        if (!isAllowed(properties)) {
+            throw new SQLException(format("Connection property '%s' is not allowed", key));
+        }
+
+        getValue(properties);
+    }
+
+    protected static final Predicate<Properties> REQUIRED = properties -> true;
+    protected static final Predicate<Properties> NOT_REQUIRED = properties -> false;
+
+    protected static final Predicate<Properties> ALLOWED = properties -> true;
+
+    interface Converter<T>
+    {
+        T convert(String value);
+    }
+
+    protected static final Converter<String> STRING_CONVERTER = value -> value;
+    protected static final Converter<File> FILE_CONVERTER = File::new;
+
+    protected static final Converter<Boolean> BOOLEAN_CONVERTER = value -> {
```

Comment:
We had that originally, but decided that since we aren't going to be 100% compatible, it's better to keep the code simple (and accepting 0/1 as a boolean is weird).